lib/core/src/wallet.rs

```diff
@@ -181,6 +206,20 @@ impl LiquidOnchainWallet {
         .map_err(|e| anyhow!("Invalid descriptor: {e}"))?;
         Ok(descriptor_str.parse()?)
     }
+
+    async fn execute_scan(&self, client: &mut LiquidClient, index: u32) -> Result<()> {
```

Comment:
Maybe change this to derivation_index too just for clarity
```suggestion
    async fn execute_scan(&self, client: &mut LiquidClient, derivation_index: u32) -> Result<()> {
```